src/main/java/twilightforest/TwilightForestMod.java

```diff
@@ -120,33 +119,73 @@ public void onInitialize() {
 		TFCreativeTabs.TABS.register();
 		TFFeatureModifiers.TREE_DECORATORS.register();
 		TFFeatureModifiers.TRUNK_PLACERS.register();
+		TFCaveCarvers.register();
+		TFEntities.init();
 
 		DwarfRabbitVariant.DWARF_RABBITS.register();
 		TinyBirdVariant.TINY_BIRDS.register();
+		registerWoodPalettes();
 		WoodPalettes.WOOD_PALETTES.register();
 		Enforcement.ENFORCEMENTS.register();
+		registerRestrictions();
 		Restrictions.RESTRICTIONS.register();
-		MagicPaintingVariants.MAGIC_PAINTINGS.register(modbus);
-
-		modbus.addListener(this::sendIMCs);
-		modbus.addListener(this::init);
-		modbus.addListener(this::registerExtraStuff);
-		modbus.addListener(this::setRegistriesForDatapack);
-		modbus.addListener(CapabilityList::registerCapabilities);
-
-		if (ModList.get().isLoaded("curios")) {
-			Bindings.getForgeBus().get().addListener(CuriosCompat::keepCurios);
-			modbus.addListener(CuriosCompat::registerCurioRenderers);
-			modbus.addListener(CuriosCompat::registerCurioLayers);
-		}
+		registerMagicPaintings();
+		MagicPaintingVariants.MAGIC_PAINTINGS.register();
 
+		Registry.register(BuiltInRegistries.BIOME_SOURCE, TwilightForestMod.prefix("twilight_biomes"), TFBiomeProvider.TF_CODEC);
+		Registry.register(BuiltInRegistries.BIOME_SOURCE, TwilightForestMod.prefix("landmarks"), LandmarkBiomeSource.CODEC);
+		Registry.register(BuiltInRegistries.CHUNK_GENERATOR, TwilightForestMod.prefix("structure_locating_wrapper"), ChunkGeneratorTwilight.CODEC);
+		init();
 		BiomeGrassColors.init();
 
-		addClassicPack();
+		if (FabricLoader.getInstance().isModLoaded("trinkets"))
+			TrinketsCompat.init();
+
 		initEvents();
-		init();
+	}
+
+	public void registerWoodPalettes() {
```

Comment:
How come these arent organized into a helper method like they are in the forge codebase?
https://github.com/TeamTwilight/twilightforest/blob/1.20.1/src/main/java/twilightforest/TwilightForestMod.java#L137

Reply:
I meant, why is it in the main mod class and not in WoodPalettes. It doesn't have anything to do with forge vs fabric. It's just an organization issue.